### PR TITLE
Use Sensible Default Configuration

### PR DESCRIPTION
**Pull request by lkiesow**
This patch changes the asset manager default configuration to use the
URL of the admin node as default with another default to the servers own
URL. This should work on both all-in-one and distributed set-ups.

This patch also changes the log level of which URL is actually used to
debug to avoid cluttering the logs with unnecessary warnings.

### CODE REVIEW

**Comment by gregorydlogan** [comment]
I can //review// this.  Seems reasonable to me.

File: modules/matterhorn-asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/OsgiEndpointHttpAssetProvider.java [hunk 1]
```diff
@@ -64,23 +65,25 @@
...
-      }
-    } else {
+    Organization organization = securityService.getOrganization();
+    if (organization == null) {
       logger.warn("No organization found! Using default server url ({})", serverUrl);
       return serverUrl;
     }
+
+    // Get asset manager URL. Default to admin node URL or to server URL
+    String orgServerUrl = organization.getProperties().get(MatterhornConstants.ASSET_MANAGER_URL_ORG_PROPERTY);
```
Comment (gregorydlogan) [inline modules/matterhorn-asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/OsgiEndpointHttpAssetProvider.java]: Do we want to trim the input here?

File: modules/matterhorn-asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/OsgiEndpointHttpAssetProvider.java [hunk 2]
```diff
@@ -64,23 +65,25 @@
...
+    if (StringUtils.isBlank(orgServerUrl)) {
+      orgServerUrl = organization.getProperties().get(MatterhornConstants.ADMIN_URL_ORG_PROPERTY);
+      logger.debug("No asset manager URL for organization '{}'. Falling back to admin node url ({})",
+                   organization, orgServerUrl);
+    }
+    if (StringUtils.isBlank(orgServerUrl)) {
+      logger.debug("No admin node URL for organization '{}' set. Falling back to default server url ({})",
+                   organization, serverUrl);
+      orgServerUrl = serverUrl;
```
Comment (gregorydlogan) [inline modules/matterhorn-asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/OsgiEndpointHttpAssetProvider.java]: Has this value already been trimmed at this point?

**Comment by lkiesow** [comment]
Closed in favor of the bitbucket pull request

**Pull request closed without merging**